packages/openid4vp/src/openid4vp-auth-response/create-openid4vp-auth-response.ts

```diff
@@ -0,0 +1,108 @@
+import { type CallbackContext, type JwtSigner, Oauth2Error } from '@openid4vc/oauth2'
+import { createJarmAuthResponse } from '../jarm/jarm-auth-response-create'
+import { extractJwksFromClientMetadata } from '../jarm/jarm-extract-jwks'
+import { jarmAssertMetadataSupported } from '../jarm/metadata/jarm-assert-metadata-supported'
+import type { JarmServerMetadata } from '../jarm/metadata/v-jarm-as-metadata'
+import type { Openid4vpAuthRequest } from '../openid4vp-auth-request/v-openid4vp-auth-request'
+import type { Openid4vpAuthResponse } from './v-openid4vp-auth-response'
+
+export async function createOpenid4vpAuthorizationResponse(options: {
+  requestParams: Pick<Openid4vpAuthRequest, 'state' | 'client_metadata' | 'nonce' | 'response_mode'>
+  responseParams: Openid4vpAuthResponse & { state?: never }
+  jarm?: {
+    jwtSigner?: JwtSigner
+    jweEncryptor?: {
+      nonce: string
+    }
+    serverMetadata: JarmServerMetadata
+    iss?: string // The issuer URL of the authorization server that created the response
+    aud?: string // The client_id of the client the response is intended for
+    exp?: number // The expiration time of the JWT. A maximum JWT lifetime of 10 minutes is RECOMMENDED.
```

Comment:
For oid4vci i usually define these as:
- `audience`
- `issuer` (or `authorizationServer` if the `iss` value is the `authorizationServer`
- `now` and `expiresInSeconds` instead of `exp`.

We should be consistent in this